Sourcery/Models/TypeName.swift

```diff
@@ -82,11 +82,21 @@ public final class TypeName: NSObject, AutoCoding, AutoEquatable, AutoDiffable,
                 self.unwrappedTypeName = name
             }
         }
+        
+        if let characters = self.unwrappedTypeName.characters.split(separator: "<").first {
+            self.baseTypeName = String(characters)
+        } else {
+            self.baseTypeName = ""
+            
+        }
     }
 
     /// Type name used in declaration
     public let name: String
-
+    
+    /// Type name without the generics
+    public var baseTypeName: String
```

Comment:
How would this work with respect to classes, enums and structs? They can be generic but aren't necessarily. I'm assuming you're on the same line where all three have a `generic: GenericType?` variable?